### PR TITLE
Connect dapp page added

### PR DESCRIPTION
**Pull request by vladdezhniuk**
## Description (optional)
Add a concise explanation of the changes.

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/scenes/plugin/ConnectDapp/Confirmation.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+  const deny = () => {
+    setConnectStep('InitialScreen')
+    window.location.replace('/#/plugin/coinview')
+  }
+  return (
+    <ConfirmationWrapper>
+      <IconsWrapper>
+        <IconBlockchainCircle width='78px' height='78px' color='#fff' />
+        <IconBlockchainCircle width='78px' height='78px' color='#fff' />
```
Comment (TarasPotsikaylo) [inline packages/blockchain-wallet-v4-frontend/src/scenes/plugin/ConnectDapp/Confirmation.tsx]: no mock color

File: packages/blockchain-wallet-v4-frontend/src/scenes/plugin/CoinsList/CoinsListHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        </Icon>
+      }
+    >
+      <WalletAddress>
+        <Text size='12px' color='grey600'>
+          <FormattedMessage id='plugin.privateKeyWallet' defaultMessage='Private Key Wallet' />
+        </Text>
+        <WalletSymbol>
+          <Text size='12px'>ETH</Text>
+          <span />
```
Comment (VitaliiShpital) [inline packages/blockchain-wallet-v4-frontend/src/scenes/plugin/CoinsList/CoinsListHeader.tsx]: interesting

File: packages/blockchain-wallet-v4-frontend/src/scenes/plugin/CoinsList/CoinsListHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        </WalletSymbol>
+      </WalletAddress>
+    </DefaultHeader>
+  )
+}
+
+const mapStateToProps = (state) => ({
+  walletAddress: selectors.core.kvStore.eth.getDefaultAddress(state).getOrElse('')
```
Comment (VitaliiShpital) [inline packages/blockchain-wallet-v4-frontend/src/scenes/plugin/CoinsList/CoinsListHeader.tsx]: is this ever used?

File: packages/blockchain-wallet-v4-frontend/src/scenes/plugin/ConnectDapp/Confirmation.tsx (hunk 1 above)
Comment (VitaliiShpital) [inline packages/blockchain-wallet-v4-frontend/src/scenes/plugin/ConnectDapp/Confirmation.tsx]: hardcoded color

File: packages/blockchain-wallet-v4-frontend/src/scenes/plugin/ConnectDapp/Confirmation.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+          <Text size='12px' lineHeight='18px' weight={500}>
+            <FormattedMessage
+              id='scenes.plugin.settings.connect_dapp.allow.funds'
+              defaultMessage='It cannot move funds without your permission'
+            />
+          </Text>
+        </AllowListItem>
+      </AllowList>
+      <Flex justifyContent='center'>
+        <DenyButton to='/plugin/coinslist' onClick={() => deny()}>
```
Comment (VitaliiShpital) [inline packages/blockchain-wallet-v4-frontend/src/scenes/plugin/ConnectDapp/Confirmation.tsx]: `onClick={deny}` should also work right?

File: packages/blockchain-wallet-v4-frontend/src/scenes/plugin/ConnectDapp/Confirmation.tsx [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+          <FormattedMessage id='scenes.plugin.settings.connect_dapp.deny' defaultMessage='Deny' />
+        </DenyButton>
+        <ConnectButton onClick={() => connect()}>
```
Comment (VitaliiShpital) [inline packages/blockchain-wallet-v4-frontend/src/scenes/plugin/ConnectDapp/Confirmation.tsx]: same about onClick

File: packages/blockchain-wallet-v4-frontend/src/scenes/plugin/ConnectDapp/index.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+const showingFrames = keyframes`
+  from { opacity: 0; transform: scale(0.1) }
+  to { opacity: 1; transform: scale(1)}
+`
+const BlockchainIcon = styled(IconBlockchain)`
+  animation: ${showingFrames} 2s ease-in-out;
+  color: ${(props) => props.theme.white};
+`
+
+export type ConnectStep = 'InitialScreen' | 'Confirmation' | 'Connectiing' | 'Connected'
```
Comment (VitaliiShpital) [inline packages/blockchain-wallet-v4-frontend/src/scenes/plugin/ConnectDapp/index.tsx]: I would use enum here. In the code below you use plain strings to set step but with enum you could use named fields

File: packages/blockchain-wallet-v4-frontend/src/scenes/plugin/CoinsList/CoinsListHeader.tsx (hunk 2 above)
Comment (vladdezhniuk) [reply packages/blockchain-wallet-v4-frontend/src/scenes/plugin/CoinsList/CoinsListHeader.tsx]: Thats not my code

Comment (vladdezhniuk) [reply packages/blockchain-wallet-v4-frontend/src/scenes/plugin/CoinsList/CoinsListHeader.tsx]: removed from pr

File: packages/blockchain-wallet-v4-frontend/src/scenes/plugin/CoinsList/CoinsListHeader.tsx (hunk 3 above)
Comment (vladdezhniuk) [reply packages/blockchain-wallet-v4-frontend/src/scenes/plugin/CoinsList/CoinsListHeader.tsx]: removed from pr, not my code

File: packages/blockchain-wallet-v4-frontend/src/scenes/plugin/ConnectDapp/index.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      setConnectStep(ConnectStep.Confirmation)
+    }, 2000)
+    return () => {
+      clearTimeout(timeout)
+    }
+  }, [])
+
+  const getConnectStep = () => {
+    switch (connectStep) {
+      case 'Confirmation':
```
Comment (VitaliiShpital) [inline packages/blockchain-wallet-v4-frontend/src/scenes/plugin/ConnectDapp/index.tsx]: `case ConnectStep.Confirmation:`

same below

**Review by NikolaiYurchenko — APPROVED** [review]

**Review by VitaliiShpital — APPROVED** [review]

**Pull request merged**